File: 1737-maximum-nesting-depth-of-the-parentheses/1737-maximum-nesting-depth-of-the-parentheses.js
/**
 * @param {string} s
 * @return {number}
 */
var maxDepth = function (s) {
    let maxNestingDepth = 0, currentDepth = 0;
    for (const ch of s) {
        if (ch === '(') {
            currentDepth++;
            maxNestingDepth = Math.max(maxNestingDepth, currentDepth);
        } else if (ch === ')') {
            currentDepth--;
        }
    }
    return maxNestingDepth;
};

File: 1737-maximum-nesting-depth-of-the-parentheses/1737-maximum-nesting-depth-of-the-parentheses.go
func maxDepth(s string) int {
	maxNestingDepth := 0
	currentDepth := 0
	for _, ch := range s {
		switch ch {
		case '(':
			currentDepth++
			if currentDepth > maxNestingDepth {
				maxNestingDepth = currentDepth
			}
		case ')':
			currentDepth--
		}
	}

	return maxNestingDepth
}